apps/obsidian/src/components/RelationshipSection.tsx

```diff
@@ -0,0 +1,231 @@
+import { TFile, Notice } from "obsidian";
+import { useState } from "react";
+import DiscourseGraphPlugin from "~/index";
+import { QueryEngine } from "~/services/QueryEngine";
+import { SearchBar } from "~/components/SearchBar";
+
+type RelationTypeOption = {
+  id: string;
+  label: string;
+  isSource: boolean;
+};
+
+type RelationshipSectionProps = {
+  plugin: DiscourseGraphPlugin;
+  activeFile: TFile;
+};
+
+export const RelationshipSection = ({
+  plugin,
+  activeFile,
+}: RelationshipSectionProps) => {
+  const [selectedRelationType, setSelectedRelationType] = useState<string>("");
+  const [selectedNode, setSelectedNode] = useState<TFile | null>(null);
+  const [isAddingRelation, setIsAddingRelation] = useState(false);
+  const queryEngine = new QueryEngine(plugin.app);
+
+  const activeNodeTypeId = (() => {
+    const fileCache = plugin.app.metadataCache.getFileCache(activeFile);
+    return fileCache?.frontmatter?.nodeTypeId;
+  })();
+
+  const getAvailableRelationTypes = () => {
+    if (!activeNodeTypeId) return [];
+
+    const options: RelationTypeOption[] = [];
+
+    const relevantRelations = plugin.settings.discourseRelations.filter(
+      (relation) =>
+        relation.sourceId === activeNodeTypeId ||
+        relation.destinationId === activeNodeTypeId,
+    );
+
+    relevantRelations.forEach((relation) => {
+      const relationType = plugin.settings.relationTypes.find(
+        (type) => type.id === relation.relationshipTypeId,
+      );
+
+      if (!relationType) return;
+
+      const isSource = relation.sourceId === activeNodeTypeId;
+
+      const existingOption = options.find(
+        (opt) => opt.id === relationType.id && opt.isSource === isSource,
+      );
+
+      if (!existingOption) {
+        options.push({
+          id: relationType.id,
+          label: isSource ? relationType.label : relationType.complement,
+          isSource,
+        });
+      }
+    });
+
+    return options;
+  };
+
+  const availableRelationTypes = getAvailableRelationTypes();
+
+  const searchNodes = async (query: string): Promise<TFile[]> => {
```

Comment:
This is being created here just to be passed to the `SearchBar`.  Let's move this to `SearchBar`.
Either pass the `activeFile` or just grab it with the `app` object.